src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -141900,9 +141900,9 @@ static void M<T>(delegate*<T, void> ptr1, delegate*<T> ptr2)
                 // (7,15): warning CS8600: Converting null literal or possible null value to non-nullable type.
                 //         T t = default;
                 Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "default").WithLocation(7, 15),
-                // (8,14): warning CS8604: Possible null reference argument for parameter '' in 'delegate*<T, void>'.
+                // (8,14): warning CS8604: Possible null reference argument for parameter 'T' in 'delegate*<T, void>'.
                 //         ptr1(t);
-                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "t").WithArguments("", "delegate*<T, void>").WithLocation(8, 14),
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "t").WithArguments("T", "delegate*<T, void>").WithLocation(8, 14),
```

Comment:
>"T" [](start = 83, length = 3)

I am not sure if it is the right thing to do to unconditionally override the format here. I understand that the behavior is somewhat better now, but it doesn't feel appropriate to refer to a function pointer parameter in this fashion. For the purpose of this diagnostics, it should be identified by position rather than type. For regular method it is identified by name, I assume.

Reply:
In a regular method it is identified by name.

So perhaps the message for a function pointer should be:

```
warning CS8604: Possible null reference argument for parameter '0' in 'delegate*<T, void>'.
```
?

Is it ok to pass in an `int` as an argument to a diagnostic which usually receives an ITypeSymbol, or do we try and keep it consistent?

Or should we create an entirely new diagnostic:

```
warning CS9999: Possible null reference argument for parameter at position '0' in 'delegate*<T, void>'.
```

Reply:
> Or should we create an entirely new diagnostic:

A new diagnostics would be ideal, I think. However, I believe this is an orthogonal issue and we probably shouldn't try to address it in this PR.

---
In reply to: [577056784](https://github.com/dotnet/roslyn/pull/51223#discussion_r577056784) [](ancestors = 577056784)